dev.exs

```diff
@@ -87,12 +97,21 @@ defmodule DemoWeb.Router do
   end
 end
 
+defmodule DemoWeb.ProxyEndpoint do
+  use Beacon.ProxyEndpoint,
+    otp_app: :beacon,
+    session_options: [store: :cookie, key: "_beacon_dev_key", signing_salt: "pMQYsz0UKEnwxJnQrVwovkBAKvU3MiuL"],
+    fallback: DemoWeb.Endpoint
+end
+
 defmodule DemoWeb.Endpoint do
   use Phoenix.Endpoint, otp_app: :beacon
 
   @session_options [store: :cookie, key: "_beacon_dev_key", signing_salt: "pMQYsz0UKEnwxJnQrVwovkBAKvU3MiuL"]
 
-  socket "/live", Phoenix.LiveView.Socket, websocket: [connect_info: [session: @session_options]]
+  # TODO: add this function in the `gen.site` generator
+  def proxy_endpoint, do: DemoWeb.ProxyEndpoint
```

Comment:
If this is supposed to represent the "default" Endpoint that Phoenix gives you, then this should go in `mix beacon.gen.proxy_endpoint` in the `update_existing_endpoints/3` because that's the only place we touch that module.

Reply:
It's the other way around actually. From the site endpoint we need to know the proxy endpoint in order to fetch the scheme and port to generate URLs. I did include that function mostly to avoid adding a new required option in `Beacon.Config`. I believe that's okay because that endpoint is generated by `gen.site` and used only by beacon sites. Sounds good?

Reply:
I think the confusion was, I thought this was supposed to represent the "default" phoenix endpoint, and not a generated beacon endpoint.  So my comment is no longer relevant 🙂